src/dcd_mapping/annotate.py

```diff
@@ -251,34 +251,35 @@ def _annotate_allele_mapping(
     post_mapped: Allele = mapped_score.post_mapped
 
     # get vrs_ref_allele_seq for pre-mapped variants
-    pre_mapped.extensions = [_get_vrs_ref_allele_seq(post_mapped, metadata, tx_results)]
-
-    # Determine reference sequence
-    if mapped_score.annotation_layer == AnnotationLayer.GENOMIC:
-        sequence_id = f"ga4gh:{mapped_score.post_mapped.location.sequenceReference.refgetAccession}"
-        accession = get_chromosome_identifier_from_vrs_id(sequence_id)
-        if accession is None:
-            raise ValueError
-        if accession.startswith("refseq:"):
-            accession = accession[7:]
-    else:
-        if tx_results is None:
-            raise ValueError  # impossible by definition
-        accession = tx_results.np
+    pre_mapped.extensions = [_get_vrs_ref_allele_seq(pre_mapped, metadata, tx_results)]
+
+    if post_mapped:
+        # Determine reference sequence
+        if mapped_score.annotation_layer == AnnotationLayer.GENOMIC:
+            sequence_id = f"ga4gh:{mapped_score.post_mapped.location.sequenceReference.refgetAccession}"
+            accession = get_chromosome_identifier_from_vrs_id(sequence_id)
+            if accession is None:
+                raise ValueError
```

Comment:
Could we add a message to these raised ValueErrors? Even if we don't expect them to occur it's easier to read what went wrong while looking at the flow.